data_pipeline/elife_article_xml/elife_article_xml_pipeline.py

```diff
@@ -64,20 +64,24 @@ def iter_unprocessed_xml_file_url_from_git_directory(
     for article_xml_url in response_json['tree']:
         if article_xml_url['size'] > 0:
             if article_xml_url['url'] not in processed_file_url_list:
-                yield article_xml_url['url']
+                yield (article_xml_url['url'], article_xml_url['path'])
 
 
-def iter_xml_file_url_and_decoded_content(
+def iter_xml_file_url_path_and_decoded_content(
     source_config: ElifeArticleXmlSourceConfig,
-    article_xml_url_list: Iterable[str]
+    article_xml_url_and_path_list: Iterable[str]
 ) -> Iterable[Tuple]:
```

Comment:
I just realised that the type annotation could be expanded, e.g. `Iterable[Tuple[str, str, str]]`.

With three fields a named tuple could also be useful, not sure if necessary here though.